File: game_server/ServerProtocol/src/CommonEnum/EChildSexType.java
package CommonEnum;

/*********
 * 子嗣性别
 **/
public enum EChildSexType {
	NONE, //0 ==== 
	BOY, //1 ==== 
	GIRL, //2 ==== 
	;
public static final EChildSexType[]  EChildSexType_Values = EChildSexType.values();
public static final int EChildSexType_Length = EChildSexType_Values.length;
public static EChildSexType EChildSexType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EChildSexType_Length){ return null; }
	return EChildSexType_Values[_ivalue];
}
}

File: game_server/ClientProtocol/CommonEnum/EChildSexType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace CommonEnum
{

/// <summary>
/// 子嗣性别
/// </summary>
public enum EChildSexType {
	NONE, //0 ==== 
	BOY, //1 ==== 
	GIRL, //2 ==== 
}

public class EChildSexTypeComparer : IEqualityComparer<EChildSexType>{
	public bool Equals(EChildSexType x, EChildSexType y) { return x == y; }
	public int GetHashCode(EChildSexType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

